### PR TITLE
Revise Role Based Access Control

### PR DESCRIPTION
**Pull request by drinkcoffee**
Revise the content for Role Based Access Control plus add some example code.

### CODE REVIEW

File: docs/framework/20sources/30implementation/access-control.md [hunk 1]
```diff
@@ -1,7 +1,56 @@
...
+accounts, to prevent attackers causing a Denial of Service attack on the contract, by
+continually pausing the contract. Using a multi-signature wallet to control this action
+is not ideal, as multiple parties would need to work together to pause the contract, thus 
+allowing attacks to continue longer than they otherwise would. In this situation, 
+multiple trusted accounts could be granted **PAUSER** role. Any one of the accounts 
+could then pause the contract.
+
+For a small project, when a contract is deployed, it might be tempting to use 
+simplistic **OWNER** style access control. 
+However, it is better to deploy a contract configured for fine grain
```
Comment (ermyas) [inline docs/framework/20sources/30implementation/access-control.md:32]: I think it is worth qualifying here that this setup, which from a security perspective is equivalent to having a single `owner` based access control, does not offer any of the security benefits of a properly configured RBAC policy discussed in the first paragraph above.

File: docs/framework/20sources/30implementation/access-control.md [hunk 2]
```diff
@@ -1,7 +1,56 @@
...
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _setupRole(PAUSER_ROLE, msg.sender);
+    }
+
+    function pause() external {
+        require(hasRole(PAUSER_ROLE, msg.sender), "Must have PAUSER role");
+        _pause();
+    }
+}
+```
```
Comment (ermyas) [inline docs/framework/20sources/30implementation/access-control.md:56]: I am not sure this example is really required, given the target audience of this document

File: docs/framework/20sources/30implementation/access-control.md [hunk 3]
```diff
@@ -1,7 +1,56 @@
 ### Role Based Access Control
-Role Based Access Control (RBAC) allows separate accounts to be 
-responsible for different actions. It allows sensitive operations 
-to be controlled by one multisig while other less sensitive 
-operations can be controlled by a different multisig. 
+[Role Based Access Control (RBAC)](https://en.wikipedia.org/wiki/Role-based_access_control) 
+allows different entites to be responsible for different configuration actions. With
```
Comment (ermyas) [inline docs/framework/20sources/30implementation/access-control.md:3]: I think that we can maybe add a sentence here that makes the key point of this section a bit more explicit, which is that a system managed by a credential with wide-ranging control potentially creates more security risks than one with narrowly-scoped privileges for different credentials and specific contexts.

**Review by ermyas — APPROVED** [review]

File: docs/framework/20sources/30implementation/access-control.md (hunk 1 above)
Comment (drinkcoffee) [reply docs/framework/20sources/30implementation/access-control.md:32]: Agreed

File: docs/framework/20sources/30implementation/access-control.md (hunk 2 above)
Comment (drinkcoffee) [reply docs/framework/20sources/30implementation/access-control.md:56]: I think we have to assume a diverse set of backgrounds, and provide as many examples as possible.

File: docs/framework/20sources/30implementation/access-control.md (hunk 3 above)
Comment (drinkcoffee) [reply docs/framework/20sources/30implementation/access-control.md]: Done

**Pull request merged**